ISSUE-7985: The topics for the "Company User" mutations have been cre… (#8104)
magento/devdocs: src/guides/v2.4/graphql/mutations/update-company-user.md

---
group: graphql
title: updateCompanyUser mutation
contributor_name: Atwix
contributor_link: https://www.atwix.com/
b2b_only: true
---

> Use the `updateCompanyUser` mutation to update the company user data.

Reviewer: ```suggestion
Use the `updateCompanyUser` mutation to update an existing company user.
```

You can get the user ID and role ID with the [`company`]({{page.baseurl}}/graphql/queries/company.html) query.

## Syntax

```graphql
mutation {
    updateCompanyUser(
        input: CompanyUserUpdateInput!
    ) {
        UpdateCompanyUserOutput
    }
}
```

## Example usage

> The following example updates the data of existing company user.

Reviewer: ```suggestion
The following example changes the job title of the specified company user.
```

**Request:**

```graphql
mutation {
  updateCompanyUser(
    input: {
      id: "Mg=="
      job_title: "Company User"
    }
  ) {
    user {
      email
      firstname
      lastname
      job_title
      telephone
      status
      role {
        id
        name
        users_count
      }
    }
  }
}
```

**Response:**

```json
{
  "data": {
    "updateCompanyUser": {
      "user": {
        "email": "jane.doe@example.com",
        "firstname": "Jane",
        "lastname": "Doe",
        "job_title": "Company User",
        "telephone": "1234567890",
        "status": "ACTIVE",
        "role": {
          "id": "MQ==",
          "name": "Default User",
          "users_count": 1
        }
      }
    }
  }
}
```

> This example assigns the company user to other role and deactivates them.

Reviewer: ```suggestion
This example deactivates the company user and assigns a different role.
```

**Request:**

```graphql
mutation {
  updateCompanyUser(
    input: {
      id: "Mg=="
      role_id: "MQ=="
      status: INACTIVE
    }
  ) {
    user {
      email
      firstname
      lastname
      job_title
      telephone
      status
      role {
        id
        name
        users_count
      }
    }
  }
}
```

**Response:**

```json
{
  "data": {
    "updateCompanyUser": {
      "user": {
        "email": "jane.doe@example.com",
        "firstname": "Jane",
        "lastname": "Doe",
        "job_title": "Company User",
        "telephone": "1234567890",
        "status": "INACTIVE",
        "role": {
          "id": "MQ==",
          "name": "Default User",
          "users_count": 1
        }
      }
    }
  }
}
```

## Input attributes

The `CompanyUserUpdateInput` input object defines the company user data.

### CompanyUserUpdateInput attributes {#CompanyUserUpdateInput}

The `CompanyUserUpdateInput` object contains the following attributes:

Attribute |  Data Type | Description
--- | --- | ---

> `id` | ID! | The encoded user ID for updating

Reviewer: ```suggestion
`id` | ID! | The encoded user ID of the company user to be updated
```

Put this attribute in alphabetic order.

> `email` | String | Company user's email address

Reviewer: ```suggestion
`email` | String | The company user's email address
```

> `firstname` | String | Company user's first name

Reviewer: ```suggestion
`firstname` | String | The company user's first name
```

> `lastname` | String | Company user's last name

Reviewer: ```suggestion
`lastname` | String | The company user's last name
```

> `job_title` | String | Company user's job title

Reviewer: ```suggestion
`job_title` | String | The company user's job title or function
```

> `role_id` | ID | Company user's role ID

Reviewer: ```suggestion
`role_id` | ID | The ID of the role assigned to the company user
```

`status` | CompanyUserStatusEnum | Indicates whether the company user is ACTIVE or INACTIVE

> `telephone` | String | Company user's phone number

Reviewer: ```suggestion
`telephone` | String | The company user's phone number
```

## Output attributes

The `UpdateCompanyUserOutput` output object contains the following attribute:

Attribute |  Data Type | Description
--- | --- | ---
`user` | Customer! | Contains company user data

### Customer attributes {#Customer}

{% include graphql/customer-output-24.md %}

## Errors

Error | Description
--- | ---

> `You do not have authorization to perform this action.` | The user with ID provided in the `input`.`id` argument not available to your company.

Reviewer: ```suggestion
`You do not have authorization to perform this action.` | The user with the ID provided in the `input`.`id` argument is not assigned to your company.
```

`No such entity with roleId = xxx` | The company role with ID `xxx` doesn't exist.

> `A customer with the same email address already exists in an associated website` | The email provided in the `input`.`email` argument has already exist to another user.

Reviewer: ```suggestion
`A customer with the same email address already exists in an associated website` | The email provided in the `input`.`email` argument belongs to another user.
```
